File: framework/cei_java/src/main/java/cn/ma/cei/impl/StringWrapper.java
package cn.ma.cei.impl;

import java.util.LinkedList;
import java.util.List;

public class StringWrapper {
    private final List<String> items = new LinkedList<>();
    private final StringBuilder result = new StringBuilder();

    public void appendStringItem(String item) {
        items.add(item);
    }

    public void combineStringItems(String separator) {
        result.setLength(0);
        items.forEach(item -> {
            if (result.length() != 0) {
                result.append(separator);
            }
            result.append(item);
        });
    }

    public String toNormalString() {
        return result.toString();
    }
}

File: framework/cei_golang/src/impl/stringwrapper.go
package impl

import (
	"strings"
)

type StringWrapper struct {
	items  []string
	result strings.Builder
}

func NewStringWrapper() *StringWrapper {
	inst := new(StringWrapper)
	return inst
}

func (inst *StringWrapper) AddStringArray(array []string, trim bool) {
	for _, value := range array {
		inst.items = append(inst.items, value)
	}
}

func (inst *StringWrapper) AppendStringItem(value string) {
	inst.items = append(inst.items, value)
}

func (inst *StringWrapper) CombineStringItems(prefix, suffix, separator string) {
	inst.result.Reset();
	for _, item := range inst.items {
		if inst.result.Len() != 0 {
			inst.result.WriteString(separator);
		}
		inst.result.WriteString(prefix + item + suffix);
	}
}

func (inst *StringWrapper) ToString() string {
	return inst.result.String()
}
